File: backend/.NET/Calindra.Desafio/Calindra.Desafio.Domain/Results/MethodResult.cs
using System.Collections.Generic;
using System.Linq;

namespace Calindra.Desafio.Domain.Results
{
    public class MethodResult : MethodResult<object>
    {
        public MethodResult()
            : base() { }

        public MethodResult(object data)
            : base(data) { }

        public MethodResult(string propertyName, params string[] messages)
            : base(propertyName, messages) { }

        public void Add(IReadOnlyCollection<Failure> failures)
        {
            foreach (var failure in failures)
                Add(failure.Property, failure.Messages.ToArray());
        }
    }
}

File: backend/Node.JS/calindra-desafio/src/shared/results/method-result.ts
import { Failure } from './failure';

export class MethodResult {
    private readonly _failures: Failure[] = [];

    constructor() { }

    public data: object;
    public erroCode: number | null = null;

    public get success(): boolean {
        return this._failures.length == 0;
    }
    public get failure(): boolean {
        return this._failures.length > 0;
    }
    public get failures(): Failure[] {
        return new Array<Failure>(...this._failures);
    }
    public get count(): number {
        return this.failures.length;
    }

    public add(propertyName: string, ...messages: string[]): MethodResult {
        propertyName = propertyName?.trim() ?? '';
        let failure = this._failures.find(f => f.propertyName.toUpperCase() === propertyName.toUpperCase());
        if (failure) {
            failure.add(...messages);
        } else {
            failure = new Failure(propertyName);
        }
        return this;
    }
}